src/x86_64/interrupt.c

```diff
@@ -323,6 +329,41 @@ void unregister_interrupt(int vector)
     interrupt_names[vector] = 0;
 }
 
+closure_function(1, 0, void, shirq_handler,
+                 list, handlers)
+{
+    list_foreach(bound(handlers), l) {
+        inthandler h = struct_from_list(l, inthandler, l);
+        int_debug("   invoking handler %s (%F)\n", h->name, h->t);
+        apply(h->t);
+    }
+}
+
+u64 allocate_shirq(void)
+{
+    u64 v = allocate_interrupt();
+    heap h = interrupt_vector_heap->meta;
```

Comment:
Sort of a nit, but could we just stash the general heap in init_interrupts rather than peek into id heap private fields?

Reply:
OK, I declared a static heap int_general variable, assigned in init_interrupts.